packages/atomic/src/components/common/refine-modal/refine-modal-common.tsx

```diff
@@ -0,0 +1,138 @@
+import {h, VNode} from '@stencil/core';
+import CloseIcon from 'coveo-styleguide/resources/icons/svg/close.svg';
+import {FacetManager, QuerySummary} from '@coveo/headless';
+import {InsightFacetManager, InsightQuerySummary} from '../../insight';
+import {InsightBindings} from '../../insight/atomic-insight-interface/atomic-insight-interface';
+import {Bindings} from '../../search/atomic-search-interface/atomic-search-interface';
+import {Button} from '../button';
+import {BaseFacetElement} from '../facets/facet-common';
+
+interface RefineModalCommonRenderProps {
+  isOpen: boolean;
+  openButton?: HTMLElement;
+}
+
+interface RefineModalCommonOptions {
+  host: HTMLElement;
+  bindings: Bindings | InsightBindings;
+  initializeQuerySummary(): QuerySummary | InsightQuerySummary;
+  onClose(): void;
+}
+
+export class RefineModalCommon {
+  private host: HTMLElement;
+  private bindings: Bindings | InsightBindings;
+  private querySummary: QuerySummary | InsightQuerySummary;
+  private onClose: () => void;
+
+  public exportparts =
+    'container,header,header-wrapper,header-ruler,body,body-wrapper,footer,footer-wrapper,footer-wrapper';
+
+  constructor(props: RefineModalCommonOptions) {
+    this.host = props.host;
+    this.bindings = props.bindings;
+    this.querySummary = props.initializeQuerySummary();
+    this.onClose = props.onClose;
+  }
+
+  public showModal() {
+    this.host.style.display = '';
```

Comment:
Not sure why the refine modal was setting its `display` attribute. I would expect `atomic-modal` to take care of everything.

Reply:
good point 🤔  let me have a look

Reply:
actually, let's have a look at that separately, I want to try and fit the full insight feature before I leave!